controllers/datadogagent/testutils/metadata_utils.go

```diff
@@ -0,0 +1,79 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package testutils_test
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CheckObjectMeta define the function signature of a check that runs again a metav1.ObjectMeta object.
+type CheckObjectMeta func(t *testing.T, obj *metav1.ObjectMeta)
+
+// CheckNamespaceName used to check if the namespace and name are the expected value on a metav1.ObjectMeta.
+func CheckNamespaceName(ns, name string) CheckObjectMeta {
+	check := func(t *testing.T, obj *metav1.ObjectMeta) {
+		if obj.Namespace != ns || obj.Name != name {
+			t.Errorf("Wrong object NS/Name, want [%s/%s], got [%s/%s]", ns, name, obj.Namespace, obj.Name)
+		}
+	}
+	return check
+}
+
+// CheckLabelIsPresent used to check if a label (key,value) is present on a metav1.ObjectMeta object.
+func CheckLabelIsPresent(key, value string) CheckObjectMeta {
+	check := func(t *testing.T, obj *metav1.ObjectMeta) {
+		checkIsPresentInMap(t, key, value, obj.Labels, "obj.Labels")
+	}
+
+	return check
+}
+
+// CheckAnnotationIsPresent used to check if an annotation (key,value) is present on a metav1.ObjectMeta object.
+func CheckAnnotationIsPresent(key, value string) CheckObjectMeta {
+	check := func(t *testing.T, obj *metav1.ObjectMeta) {
+		checkIsPresentInMap(t, key, value, obj.Annotations, "obj.Annotations")
+	}
+
+	return check
+}
+
+// CheckLabelIsNotPresent used to check if a label (key,value) is not present on a metav1.ObjectMeta object.
+func CheckLabelIsNotPresent(key string) CheckObjectMeta {
```

Comment:
As the functions return the `CheckObjectMeta` meta type, we cannot easily combine them, e.g. wrapping `CheckLabelIsPresent` in a `not` or something, meaning that we'll have to provide a helper each time (regexp, not regexp, etc.)

Reply:
Do you see a solution?

Reply:
Maybe we could change to `func Check(runtime.Object) bool` and use a typed struct to inject other parameters.
Something like:
```
CheckLabelIsPresent{label: <some_key>}
with func (*CheckLabelIsPresent) Check(runtime.Object) bool
```

Reply:
@vboulineau done